core/src/main/scala/org/apache/spark/executor/TaskMetrics.scala

```diff
@@ -326,36 +324,13 @@ private[spark] object TaskMetrics extends Logging {
 
 
 private[spark] class BlockStatusesAccumulator
-  extends AccumulatorV2[(BlockId, BlockStatus), java.util.List[(BlockId, BlockStatus)]] {
-  private val _seq = Collections.synchronizedList(new ArrayList[(BlockId, BlockStatus)]())
-
-  override def isZero(): Boolean = _seq.isEmpty
+  extends CollectionAccumulator[(BlockId, BlockStatus)] {
 
   override def copyAndReset(): BlockStatusesAccumulator = new BlockStatusesAccumulator
 
   override def copy(): BlockStatusesAccumulator = {
     val newAcc = new BlockStatusesAccumulator
-    newAcc._seq.addAll(_seq)
+    newAcc._list.addAll(_list)
```

Comment:
Don't you just want to call merge() here? it's better not to expose the internal representation even to subclasses if possible, because we're exposing it permanently to everyone.